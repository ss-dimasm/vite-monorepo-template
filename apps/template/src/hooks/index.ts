/* istanbul ignore file */
export * from './use-platform-get'
export * from './use-platform-update'
export * from './utils'
