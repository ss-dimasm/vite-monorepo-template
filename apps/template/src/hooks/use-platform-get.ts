import { useEffect, useMemo } from 'react'
import {
  StringMap,
  getMergedHeaders,
  handleReapitError,
  getUrl,
  RC_SESSION_MISSING_ERROR,
  NETWORK_ERROR,
} from './utils'
import { useSnack } from '@reapit/elements'
import { useQuery, useQueryClient } from '@tanstack/react-query'
import axios, { AxiosError } from 'axios'
import { NavigateFunction, useNavigate } from 'react-router'
import { reapitConnectBrowserSession } from '../core/connect-session'

export type PlatformGet<DataType> = [
  data: DataType | null,
  loading: boolean,
  error: string | null,
  refresh: (queryParams?: Object) => void,
  refreshing: boolean,
  clearCache: () => void,
]

export interface UsePlatformGetParams {
  path: string
  errorMessage?: string
  successMessage?: string
  queryParams?: Object
  headers?: StringMap
  fetchWhenTrue?: any[]
  retry?: number | boolean
  onSuccess?: (message: string) => void
  onError?: (message: string) => void
}

export const handleError =
  (
    isError: boolean,
    error: AxiosError<any, any> | null,
    errorSnack: (text: string, timeout?: number | undefined) => void,
    navigate: NavigateFunction,
    errorMessage?: string,
    onError?: (message: string) => void,
  ) =>
  () => {
    if (isError && error) {
      const isRcError = error.message === RC_SESSION_MISSING_ERROR
      const isFourOOne = error.code === NETWORK_ERROR
      if (isRcError || isFourOOne) {
        return navigate('/login')
      }

      const errorString = handleReapitError(error, errorMessage)
      if (onError && errorString) onError(errorString)
      if (!onError && errorString) errorSnack(errorString)
      console.error(errorString)
    }
  }

export const handleSuccess =
  (
    isSuccess: boolean,
    successSnack: (text: string, timeout?: number | undefined) => void,
    successMessage?: string,
    onSuccess?: (message: string) => void,
  ) =>
  () => {
    if (isSuccess) {
      if (onSuccess && successMessage) onSuccess(successMessage)
      if (!onSuccess && successMessage) successSnack(successMessage)
    }
  }

export const usePlatformGet = <DataType>({
  path,
  errorMessage,
  successMessage,
  queryParams,
  headers,
  fetchWhenTrue,
  onSuccess,
  onError,
  retry,
}: UsePlatformGetParams): PlatformGet<DataType> => {
  const { success: successSnack, error: errorSnack } = useSnack()
  const navigate = useNavigate()
  const queryClient = useQueryClient()

  const isEnabled = fetchWhenTrue?.length
    ? Boolean(fetchWhenTrue?.filter(Boolean).length === fetchWhenTrue.length)
    : true

  const url = useMemo(getUrl(path, queryParams), [path, queryParams])

  const { data, error, isLoading, isSuccess, isError, isRefetching, refetch } = useQuery<DataType, AxiosError<any>>({
    queryKey: [url],
    queryFn: async () => {
      const reqHeaders = await getMergedHeaders(reapitConnectBrowserSession, headers)

      if (!reqHeaders) throw new Error(RC_SESSION_MISSING_ERROR)

      const req = await axios.get<DataType>(url, {
        headers: reqHeaders,
      })
      return req.data
    },
    retry: retry === undefined ? 1 : retry,
    refetchOnWindowFocus: false,
    enabled: isEnabled,
  })

  useEffect(handleSuccess(isSuccess, successSnack, successMessage, onSuccess), [
    isSuccess,
    successMessage,
    onSuccess,
    successSnack,
  ])

  useEffect(handleError(isError, error, errorSnack, navigate, errorMessage, onError), [
    isError,
    error,
    errorMessage,
    onError,
    errorSnack,
    navigate,
  ])

  const result = data ? data : null
  const errorString = error?.message ? error.message : null
  const loading = isEnabled && isLoading
  const clearCache = () => {
    queryClient.removeQueries({ queryKey: [url] })
  }

  return [result, loading, errorString, refetch, isRefetching, clearCache]
}
