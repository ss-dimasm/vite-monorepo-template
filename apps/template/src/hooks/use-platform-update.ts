import { useMemo } from 'react'
import { useSnack } from '@reapit/elements'
import {
  getMergedHeaders,
  getUrl,
  handleReapitError,
  NETWORK_ERROR,
  RC_SESSION_MISSING_ERROR,
  StringMap,
} from './utils'
import { useMutation } from '@tanstack/react-query'
import axios, { AxiosError, AxiosResponseHeaders } from 'axios'
import { useNavigate } from 'react-router'
import { reapitConnectBrowserSession } from '../core/connect-session'

export type ReapitUpdateState<ParamsType, DataType> = [
  UpdateFunction<ParamsType, DataType>,
  boolean,
  DataType | null,
  boolean,
  boolean,
]

type AcceptedMethod = 'POST' | 'PUT' | 'PATCH' | 'DELETE' | 'GET'

export interface ReapitUpdate {
  path: string
  method?: AcceptedMethod
  shouldReturnRecord?: boolean
  headers?: StringMap
  errorMessage?: string
  successMessage?: string
}

export type UpdateFunction<ParamsType, DataType> = (params: ParamsType) => Promise<DataType>

export const usePlatformUpdate = <ParamsType, DataType>({
  path,
  method = 'POST',
  shouldReturnRecord = false,
  headers = {},
  errorMessage,
  successMessage,
}: ReapitUpdate): ReapitUpdateState<ParamsType, DataType> => {
  const { error: errorSnack, success: successSnack } = useSnack()
  const navigate = useNavigate()
  const url = useMemo(getUrl(path), [path])

  const { mutateAsync, data, isSuccess, isError, isPending } = useMutation<DataType, AxiosError<any>, ParamsType>({
    mutationKey: [url],
    mutationFn: async (data: ParamsType) => {
      const updateHeaders = await getMergedHeaders(reapitConnectBrowserSession, headers)

      if (!updateHeaders) throw new Error(RC_SESSION_MISSING_ERROR)

      const res = await axios<DataType>(url, {
        method,
        headers: updateHeaders,
        data,
      })

      if (!shouldReturnRecord) return true

      const resHeaders = res.headers as AxiosResponseHeaders
      const location = resHeaders.get('Location')?.valueOf() as string

      if (!location) throw new Error('Location was not returned by server')

      const locationUrl = location.includes('.prod.paas') ? location.replace('.prod.paas', '') : location

      const locationRes = await axios(locationUrl, {
        method: 'GET',
        headers: updateHeaders,
      })

      return locationRes.data
    },
    onSuccess: () => {
      if (successMessage) successSnack(successMessage)
    },
    onError: (error: AxiosError<any>) => {
      const isRcError = error?.message === RC_SESSION_MISSING_ERROR
      const isFourOOne = error.code === NETWORK_ERROR

      if (isRcError || isFourOOne) {
        return navigate('/login')
      }

      const errorString = !isRcError ? handleReapitError(error, errorMessage) : null
      if (errorString) errorSnack(errorString, 5000)
      console.error(errorString)
    },
  })

  const updateFunction: UpdateFunction<ParamsType, DataType> = (data: ParamsType) => mutateAsync(data)
  const returnData = data ?? null

  return [updateFunction, isPending, returnData, isSuccess, isError]
}
