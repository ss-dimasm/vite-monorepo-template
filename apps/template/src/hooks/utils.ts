import { ReapitConnectBrowserSession } from '@reapit/connect-session'
import { AxiosError } from 'axios'
import qs from 'qs'

export const RC_SESSION_MISSING_ERROR =
  'Missing valid Reapit Connect Session, please try logging in again if the problem persists'
export const NETWORK_ERROR = 'ERR_NETWORK'

export interface StringMap {
  [key: string]: string
}

export interface ReapitErrorField {
  field?: string
  message?: string
}

export interface ReapitError {
  statusCode?: number
  errors?: ReapitErrorField[]
  description?: string
  dateTime?: string
  message?: string
  title?: string
}

export const getMergedHeaders = async (
  reapitConnectBrowserSession: ReapitConnectBrowserSession,
  headers?: StringMap,
): Promise<StringMap | null> => {
  const connectSession = await reapitConnectBrowserSession.connectSession()
  const accessToken = connectSession?.accessToken

  return accessToken
    ? {
        Authorization: `Bearer ${accessToken}`,
        'api-version': 'latest',
        'Content-Type': 'application/json',
        ...headers,
      }
    : null
}

export const handleReapitError = (error: AxiosError<any>, defaultMessage?: string): string => {
  const reapitError: ReapitError = error?.response?.data
  const { description, title, message, errors } = reapitError ?? {}
  const messageString = description
    ? description
    : title
      ? title
      : message
        ? message
        : error?.message
          ? error.message
          : defaultMessage
            ? defaultMessage
            : 'An unknown error has occurred, please refresh the page and try again.'
  const fieldErrors = Array.isArray(errors) ? errors?.map(({ field, message }) => `"${field}: ${message}"`) : null
  const fieldString = fieldErrors ? fieldErrors.join(', ') : ''

  return `${messageString} ${fieldString}`
}

export const stringListToBatchQuery = (list: (string | number)[], queryKey: string): string =>
  list.reduce((query: string, nextItem: string | number, index: number) => {
    if (!index) {
      return String(nextItem)
    }
    return `${query}&${queryKey}=${nextItem}`
  }, '')

// Where you have an object of filters with a mixture of arrays and strings / boolean / numeric values and
// you need a string map returned for useReapitGet
export const objectToQuery = <QueryObjectType extends {}>(queryObject: QueryObjectType): StringMap =>
  Object.keys(queryObject).reduce((currentQuery: StringMap, nextItem: string) => {
    const objectItem = queryObject[nextItem]

    if (Array.isArray(objectItem)) {
      currentQuery[nextItem] = stringListToBatchQuery(objectItem, nextItem)
    } else if (objectItem) {
      currentQuery[nextItem] = String(objectItem)
    }

    return currentQuery
  }, {} as StringMap)

export const getUrl = (path: string, queryParams?: Object) => (): string => {
  const api = process.env.PLATFORM_API_URL
  const normalisedQuery = objectToQuery(queryParams ?? {})
  const query = qs.stringify(normalisedQuery, { encode: false })
  const url = `${api}${path}${query ? `?${query}` : ''}`
  return url
}
