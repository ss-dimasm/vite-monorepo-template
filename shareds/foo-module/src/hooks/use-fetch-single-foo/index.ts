interface UseFetchSingleFooOptions {
  embedBaz?: boolean
}

export const useFetchSingleFoo = (fooId: string, options: UseFetchSingleFooOptions) => {
  const { embedBaz } = options ?? {}

  return {
    data: fooId,
    isLoading: false,
    metadata: {
      bazEntity: embedBaz ? 'baz-entity' : null,
    },
  }
}
