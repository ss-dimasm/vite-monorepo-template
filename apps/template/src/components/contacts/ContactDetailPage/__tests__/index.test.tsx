import { ChangeEvent } from 'react'
import { ContactDetailPage, handleChangeTab, handleResetForm, handleSubmitContact } from '..'
import type { ContactFormSchema } from '../../common/ContactForm/validationSchema'
import { usePlatformGet } from '../../../../hooks'
import { mockContactModel } from '../../../../scripts/stubs'
import { render } from '../../../../scripts/tests'
import { Mock } from 'vitest'

vi.mock('../../../../hooks', () => ({
  usePlatformGet: vi.fn(),
  usePlatformUpdate: vi.fn(() => [vi.fn()]),
}))

const mockUsePlatformGet = usePlatformGet as Mock

describe('ContactDetailPage', () => {
  it('should match a snapshot with data', () => {
    mockUsePlatformGet.mockReturnValue([mockContactModel, false])

    expect(render(<ContactDetailPage />).asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot when loading', () => {
    mockUsePlatformGet.mockReturnValue([null, true])

    expect(render(<ContactDetailPage />).asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot when not loading but no contact', () => {
    mockUsePlatformGet.mockReturnValue([null, false])

    expect(render(<ContactDetailPage />).asFragment()).toMatchSnapshot()
  })
})

describe('handleChangeTab', () => {
  it('should validate the personal tab then navigate', async () => {
    window.location.pathname = '/personal'
    const trigger = vi.fn()
    const navigate = vi.fn()
    const contactId = 'FOO'
    const event = { target: { value: 'foo' } } as ChangeEvent<HTMLInputElement>

    const curried = handleChangeTab(trigger, navigate, contactId)

    curried(event)

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith(['title', 'dateOfBirth', 'forename', 'surname'])
    expect(navigate).toHaveBeenCalledWith('/contacts/FOO/foo')
  })

  it('should validate the communications tab then navigate', async () => {
    window.location.pathname = '/communications'
    const trigger = vi.fn()
    const navigate = vi.fn()
    const contactId = 'FOO'
    const event = { target: { value: 'foo' } } as ChangeEvent<HTMLInputElement>

    const curried = handleChangeTab(trigger, navigate, contactId)

    curried(event)

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith([
      'homePhone',
      'workPhone',
      'mobilePhone',
      'email',
      'marketingConsent',
      'communicationPreferenceEmail',
      'communicationPreferencePhone',
      'communicationPreferenceLetter',
      'communicationPreferenceSMS',
    ])

    expect(navigate).toHaveBeenCalledWith('/contacts/FOO/foo')
  })

  it('should validate the addresses tab then navigate', async () => {
    window.location.pathname = '/addresses'
    const trigger = vi.fn()
    const navigate = vi.fn()
    const contactId = 'FOO'
    const event = { target: { value: 'foo' } } as ChangeEvent<HTMLInputElement>

    const curried = handleChangeTab(trigger, navigate, contactId)

    curried(event)

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith(['primaryAddress', 'secondaryAddress', 'workAddress'])

    expect(navigate).toHaveBeenCalledWith('/contacts/FOO/foo')
  })

  it('should validate the office tab then navigate', async () => {
    window.location.pathname = '/office'
    const trigger = vi.fn()
    const navigate = vi.fn()
    const contactId = 'FOO'
    const event = { target: { value: 'foo' } } as ChangeEvent<HTMLInputElement>

    const curried = handleChangeTab(trigger, navigate, contactId)

    curried(event)

    await Promise.resolve()

    expect(trigger).toHaveBeenCalledWith(['negotiatorIds', 'categoryIds', 'officeIds', 'source'])

    expect(navigate).toHaveBeenCalledWith('/contacts/FOO/foo')
  })
})

describe('handleSubmitContact', () => {
  it('should submit a contact then navigate', async () => {
    window.location.pathname = '/personal'
    const updateContact = vi.fn(() => Promise.resolve(true))
    const navigate = vi.fn()
    const values = {
      surname: 'MOCK_NAME',
      categoryIds: 'MOCK_CATEGORY',
      negotiatorIds: 'MOCK_NEG',
      officeIds: 'MOCK_OFFICE',
      source: 'MOCK-SOURCE',
    } as ContactFormSchema

    const curried = handleSubmitContact(updateContact, navigate)

    await curried(values)

    expect(updateContact).toHaveBeenCalledWith({
      ...values,
      categoryIds: ['MOCK_CATEGORY'],
      metadata: undefined,
      negotiatorIds: ['MOCK_NEG'],
      officeIds: ['MOCK_OFFICE'],
      source: {
        id: 'MOCK',
        type: 'SOURCE',
      },
    })
    expect(navigate).toHaveBeenCalledWith('/contacts')
  })
})

describe('handleResetForm', () => {
  it('should submit a contact then navigate', () => {
    window.location.pathname = '/personal'
    const reset = vi.fn()

    const curried = handleResetForm(mockContactModel, reset)

    curried()

    expect(reset).toHaveBeenCalledWith({
      ...mockContactModel,
      source: mockContactModel?.source ? `${mockContactModel.source.id}-${mockContactModel.source.type}` : '',
      categoryIds: mockContactModel?.categoryIds?.join(','),
      negotiatorIds: mockContactModel?.negotiatorIds?.join(','),
      officeIds: mockContactModel?.officeIds?.join(','),
    })
  })
})
