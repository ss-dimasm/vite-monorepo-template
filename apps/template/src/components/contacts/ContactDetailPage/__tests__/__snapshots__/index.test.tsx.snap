// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactDetailPage > should match a snapshot when loading 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
  <div
    class="el-flex-container el-flex-auto"
  >
    <section
      class="el-page-container el-has-grey-background"
    >
      <div
        class="mocked-styled-178 el-page-header-wrap"
      >
        <div
          class="el-page-header-max-width el-page-header-wrap-inner"
        >
          <div
            class="mocked-styled-175 el-page-header-container"
          >
            <div
              class="mocked-styled-175 el-page-header-container"
            >
              <div
                class="el-flex-container el-flex-column"
              >
                <div
                  class="mocked-styled-176 el-page-header-title-container"
                >
                  <h1
                    class="el-text-base el-text2xl el-has-bold-text el-mr3"
                  >
                    Unknown
                  </h1>
                </div>
              </div>
            </div>
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="mocked-styled-3 el-button-group-inner"
              >
                <button
                  class="el-button-size-small el-intent-neutral el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Back To List
                </button>
                <button
                  class="el-button-size-small el-intent-primary el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  New Contact
                </button>
              </div>
            </div>
          </div>
          <div
            class="mocked-styled-119 el-tabs-wrap"
          >
            <div
              class="mocked-styled-120 el-tabs-options-wrap"
              role="tablist"
            >
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="overview"
                name="app-edit-tabs"
                type="radio"
                value="overview"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="overview"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Overview
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="personal"
                name="app-edit-tabs"
                type="radio"
                value="personal"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="personal"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Personal
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="communications"
                name="app-edit-tabs"
                type="radio"
                value="communications"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="communications"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Communications
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="addresses"
                name="app-edit-tabs"
                type="radio"
                value="addresses"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="addresses"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Addresses
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="office"
                name="app-edit-tabs"
                type="radio"
                value="office"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="office"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Office Details
                </span>
              </label>
            </div>
            <div
              class="el-tabs-has-no-border el-tabs-footer"
            />
          </div>
        </div>
      </div>
      <div
        class="el-flex-container el-flex-column el-has-max-width"
      >
        <div
          class="mocked-styled-172 el-tile"
        >
          <div
            class="mocked-styled-44 el-loader-container"
          >
            <div
              class="mocked-styled-46 el-loader"
            >
              <div
                class="mocked-styled-47 el-loader-moving-bar"
              />
            </div>
          </div>
          <form>
            <div
              class="el-mb7"
            />
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="el-button-group-align-left el-button-group-inner"
              >
                <button
                  class="el-intent-primary el-button"
                  disabled=""
                  type="button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Update
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`ContactDetailPage > should match a snapshot when not loading but no contact 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
  <div
    class="el-flex-container el-flex-auto"
  >
    <section
      class="el-page-container el-has-grey-background"
    >
      <div
        class="mocked-styled-178 el-page-header-wrap"
      >
        <div
          class="el-page-header-max-width el-page-header-wrap-inner"
        >
          <div
            class="mocked-styled-175 el-page-header-container"
          >
            <div
              class="mocked-styled-175 el-page-header-container"
            >
              <div
                class="el-flex-container el-flex-column"
              >
                <div
                  class="mocked-styled-176 el-page-header-title-container"
                >
                  <h1
                    class="el-text-base el-text2xl el-has-bold-text el-mr3"
                  >
                    Unknown
                  </h1>
                </div>
              </div>
            </div>
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="mocked-styled-3 el-button-group-inner"
              >
                <button
                  class="el-button-size-small el-intent-neutral el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Back To List
                </button>
                <button
                  class="el-button-size-small el-intent-primary el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  New Contact
                </button>
              </div>
            </div>
          </div>
          <div
            class="mocked-styled-119 el-tabs-wrap"
          >
            <div
              class="mocked-styled-120 el-tabs-options-wrap"
              role="tablist"
            >
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="overview"
                name="app-edit-tabs"
                type="radio"
                value="overview"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="overview"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Overview
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="personal"
                name="app-edit-tabs"
                type="radio"
                value="personal"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="personal"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Personal
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="communications"
                name="app-edit-tabs"
                type="radio"
                value="communications"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="communications"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Communications
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="addresses"
                name="app-edit-tabs"
                type="radio"
                value="addresses"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="addresses"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Addresses
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="office"
                name="app-edit-tabs"
                type="radio"
                value="office"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="office"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Office Details
                </span>
              </label>
            </div>
            <div
              class="el-tabs-has-no-border el-tabs-footer"
            />
          </div>
        </div>
      </div>
      <div
        class="el-flex-container el-flex-column el-has-max-width"
      >
        <div
          class="mocked-styled-172 el-tile"
        >
          <form>
            <div
              class="el-mb7"
            />
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="el-button-group-align-left el-button-group-inner"
              >
                <button
                  class="el-intent-primary el-button"
                  disabled=""
                  type="button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Update
                </button>
              </div>
            </div>
          </form>
          <div
            class="el-intent-primary el-is-active el-pn-is-full-width el-pn-is-inline el-persistent-notification"
            role="status"
          >
            <div
              class="el-pn-icon"
              data-testid="close-icon"
            >
              <span
                class="el-icon-new el-icon"
                style="font-size: 1.25rem;"
              >
                <svg
                  fill="none"
                  height="1em"
                  role="img"
                  style="font-size: 1.25rem;"
                  title="Icon image with name info"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10A10 10 0 0 0 12 2m1.143 15.714h-2.286v-6h2.286zm-1.143-8a1.714 1.714 0 1 1 0-3.428 1.714 1.714 0 0 1 0 3.428"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
            <div
              aria-live="polite"
              class="el-pn-content"
            >
              Contact not found
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`ContactDetailPage > should match a snapshot with data 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
  <div
    class="el-flex-container el-flex-auto"
  >
    <section
      class="el-page-container el-has-grey-background"
    >
      <div
        class="mocked-styled-178 el-page-header-wrap"
      >
        <div
          class="el-page-header-max-width el-page-header-wrap-inner"
        >
          <div
            class="mocked-styled-175 el-page-header-container"
          >
            <div
              class="mocked-styled-175 el-page-header-container"
            >
              <div
                class="el-flex-container el-flex-column"
              >
                <div
                  class="mocked-styled-176 el-page-header-title-container"
                >
                  <h1
                    class="el-text-base el-text2xl el-has-bold-text el-mr3"
                  >
                    Mr Foo Bar
                  </h1>
                </div>
              </div>
            </div>
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="mocked-styled-3 el-button-group-inner"
              >
                <button
                  class="el-button-size-small el-intent-neutral el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Back To List
                </button>
                <button
                  class="el-button-size-small el-intent-primary el-button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  New Contact
                </button>
              </div>
            </div>
          </div>
          <div
            class="mocked-styled-119 el-tabs-wrap"
          >
            <div
              class="mocked-styled-120 el-tabs-options-wrap"
              role="tablist"
            >
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="overview"
                name="app-edit-tabs"
                type="radio"
                value="overview"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="overview"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Overview
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="personal"
                name="app-edit-tabs"
                type="radio"
                value="personal"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="personal"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Personal
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="communications"
                name="app-edit-tabs"
                type="radio"
                value="communications"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="communications"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Communications
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="addresses"
                name="app-edit-tabs"
                type="radio"
                value="addresses"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="addresses"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Addresses
                </span>
              </label>
              <input
                aria-selected="false"
                class="mocked-styled-117 el-tabs"
                id="office"
                name="app-edit-tabs"
                type="radio"
                value="office"
              />
              <label
                class="mocked-styled-118 el-tabs-label"
                for="office"
                role="tab"
                tabindex="0"
              >
                <span
                  class="el-tabs-item"
                >
                  Office Details
                </span>
              </label>
            </div>
            <div
              class="el-tabs-has-no-border el-tabs-footer"
            />
          </div>
        </div>
      </div>
      <div
        class="el-flex-container el-flex-column el-has-max-width"
      >
        <div
          class="mocked-styled-172 el-tile"
        >
          <form>
            <div
              class="el-mb7"
            />
            <div
              class="el-mb7 el-intent-danger el-is-active el-pn-is-full-width el-pn-is-inline el-persistent-notification"
              role="status"
            >
              <div
                class="el-pn-icon"
                data-testid="close-icon"
              >
                <span
                  class="el-icon-new el-icon"
                  style="font-size: 1.25rem;"
                >
                  <svg
                    fill="none"
                    height="1em"
                    role="img"
                    style="font-size: 1.25rem;"
                    title="Icon image with name info"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10A10 10 0 0 0 12 2m1.143 15.714h-2.286v-6h2.286zm-1.143-8a1.714 1.714 0 1 1 0-3.428 1.714 1.714 0 0 1 0 3.428"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <div
                aria-live="polite"
                class="el-pn-content"
              >
                There are one or more errors in the form that are preventing you from saving the contact. Please check each section to proceeed.
              </div>
            </div>
            <div
              class="mocked-styled-2 el-button-group"
            >
              <div
                class="el-button-group-align-left el-button-group-inner"
              >
                <button
                  class="el-intent-primary el-button"
                  disabled=""
                  type="button"
                >
                  <div
                    class="mocked-styled-0 el-button-loader"
                  />
                  Update
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
